journalpump/journalpump.py

```diff
@@ -675,6 +723,16 @@ def _truncate_long_message(self, json_entry):
         }
         return json.dumps(entry, default=default_json_serialization).encode("utf8")
 
+    def _apply_secret_filters(self, data):
+        msg = data.get("MESSAGE")
+        if msg:
+            for secret_filter in self.reader.secret_filters:
+                if self.reader.secret_filter_metrics is True:
+                    self.reader.secret_filter_matches = self.reader.secret_filter_matches + 1
```

Comment:
Nothing checks here that there in fact was a match. `re.sub` doesn't directly give that info but I guess we could just see if new message is different from original message to determine that.

Also, nit: `+=` would be shorter.

Reply:
Thanks again for taking a look Rauli. I decided to remove the "if match" for 2 reasons. 

1. the re.sub operation returns the string unchanged if there is no match, so we get the same behavior in much simpler structure, with only 1 regex search instead of 2 regex search operations. 
2. with the simpler regex pattern, re.match only works if the pattern is at the start of the string... I could swap it for something like re.search, but decided against it for reason #1. 

Also, shortened to +=  now.

Reply:
I didn't quite follow. My point was that this will now report metric data point for every log line, no matter whether it matches or not. I suggested having something like:

```
original_msg = msg
msg = pattern.sub(...)
if original_msg != msg:
    self.reader.secret_filter_matches += 1
```

Btw. when writing that line there I realized the counter will be incremented for every sender so if you have multiple senders the count is multiplied by the number of those.

Reply:
ahh, the first was an oversight when i was refactoring. Fixed that, but.. i hadn't noticed this code was called for every sender. I'll have to think about what to do to resolve that.

Reply:
I moved the _apply_filters method up to process(), so now it is run only once ( per line per reader)